src/main/java/org/cactoos/io/TempFile.java

```diff
@@ -0,0 +1,150 @@
+import org.cactoos.Scalar;
+import org.cactoos.Text;
+import org.cactoos.scalar.IoCheckedScalar;
+import org.cactoos.scalar.StickyScalar;
+import org.cactoos.text.TextOf;
+
+/**
+ * A temporary file.
+ * <p>
+ * These are ephemeral files to be used in small scopes.
+ * Typical use looks like this:
+ * <pre>{@code
+ *    try (final TempFile file = new TempFile()) {
+ *       //write to the file
+ *    }
+ * }</pre>
+ * The physical file is deleted from the filesystem when the temp file is
+ * closed.
+ *
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public final class TempFile implements Scalar<Path>, Closeable {
+    /**
+     * Creates the temporary file, returning its path.
+     */
+    private final Scalar<Path> file;
+
+    /**
+     * Ctor.
+     * <p>
+     * Specifies empty strings for suffix and prefix, and creates the file
+     * in the filesystem's temporary directory denoted by the system property
+     * {@code java.io.tmpdir}.
+     * @since 1.0
+     */
+    public TempFile() {
+        this("", "");
+    }
+
+    /**
+     * Ctor.
+     * <p>
+     * The temporary file will be created inside the filesystem's
+     * temporary folder (system property: {@code java.io.tmpdir}).
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final String prefix,
+        final String suffix) {
+        this(
+            () -> Paths.get(System.getProperty("java.io.tmpdir")),
+            prefix,
+            suffix
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param dir The directory in which to create the temp file
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final Scalar<Path> dir,
+        final String prefix,
+        final String suffix) {
+        this(
+            dir,
+            new TextOf(prefix),
+            new TextOf(suffix)
+        );
+    }
+
+    /**
+     * Ctor.
+     * @param dir The directory in which to create the temp file
+     * @param prefix The temp filename's prefix
+     * @param suffix The temp filename's suffix
+     * @since 1.0
+     */
+    public TempFile(
+        final Scalar<Path> dir,
+        final Text prefix,
+        final Text suffix) {
+        this(
+            new StickyScalar<>(
+                () -> Files.createTempFile(
+                    dir.value(),
+                    prefix.asString(),
+                    suffix.asString()
+                )
+            )
+        );
+    }
+
+    /**
+     * Primary ctor.
+     * @param fullpath Creates the file and returns the path to it
+     * @since 1.0
+     */
+    private TempFile(final Scalar<Path> fullpath) {
+        this.file = fullpath;
+    }
+
+    @Override
+    public Path value() throws Exception {
+        return this.file.value();
+    }
+
+    @Override
```

Comment:
@llorllale Add a JavaDoc to this method, to make it clear that it will actually delete the file.